packages/slate-tables/src/normalization/normalizeTableData.ts

```diff
@@ -0,0 +1,89 @@
+import { omit } from 'lodash-es';
+import type { Path } from 'slate';
+import { Node, Transforms } from 'slate';
+import { Editor } from 'slate';
+
+import { TablesEditor } from '../TablesEditor';
+
+export function normalizeTableData(editor: TablesEditor, path: Path) {
```

Comment:
If I'm not mistaken, this normalization splits merged cells (with `colspan` / `rowspan` attributes) into single 1x1 cells, right?

If so, I'd recommend renaming the function to make that obvious. `normalizeTableData` is too generic and can be used for any other normalization introduced in this PR.

How about `splitMergedCells`?

I also think that this function is too big and too complex to understand. I'd suggest splitting it into two: 

- `splitColSpanCells`
- `splitRowSpanCells`

```ts
export function splitColSpanCells(editor: TablesEditor, path: Path) {
    const node = Node.get(editor, path);

    if (editor.isTableCellNode(node) && node.colspan && node.colspan > 1) {
        const padCells = times(node.colspan - 1).map(() => TablesEditor.createTableCell(editor));

        Editor.withoutNormalizing(editor, () => {
            Transforms.unsetNodes<TableCellNode>(editor, 'colspan', { at: path });
            Transforms.insertNodes(editor, padCells, { at: Path.next(path) });
        });
        return true;
    }

    return false;
}

export function splitRowSpanCells(editor: TablesEditor, path: Path) {
    const node = Node.get(editor, path);

    if (editor.isTableNode(node)) {
        for (const [cell, cellPath] of TablesEditor.cells(editor, [node, path])) {
            if (cell.rowspan && cell.rowspan > 1) {
                const [rowIndex, cellIndex] = cellPath.slice(-2);
                const padCells = times(cell.rowspan - 1).map(() =>
                    TablesEditor.createTableCell(editor),
                );

                Editor.withoutNormalizing(editor, () => {
                    Transforms.unsetNodes<TableCellNode>(editor, 'rowspan', { at: cellPath });
                    for (const [offset, padCell] of padCells.entries()) {
                        // TODO: Might be worth checking if the insertion path exists.
                        Transforms.insertNodes(editor, padCell, {
                            at: [...path, rowIndex + 1 + offset, cellIndex],
                        });
                    }
                });

                return true;
            }
        }
    }

    return false;
}
```

Reply:
i am not sure that it is possible to make two different functions as i have to restore cells in on loop

Reply:
> as i have to restore cells in one loop

Hm.. Why?